csharp/src/Drivers/Apache/Hive2/IHiveServer2Statement.cs

```diff
@@ -40,7 +34,9 @@ internal interface IHiveServer2Statement : ITracingStatement
         /// Checks if direct results are available.
         /// </summary>
         /// <returns>True if direct results are available and contain result data, false otherwise.</returns>
-        bool HasDirectResults { get; }
+        bool HasDirectResults(IResponse response);
+
+        bool TryGetDirectResults(IResponse response, out TSparkDirectResults? directResults);
```

Comment:
Given that all the other methods in the interface have doc comments, could you add them here too?